openwisp-config/files/sbin/openwisp-update-config.lua

```diff
@@ -148,8 +148,26 @@ if lfs.attributes(remote_config_dir, 'mode') == 'directory' then
         -- ensure we are acting on a file
         if lfs.attributes(remote_path, 'mode') == 'file' then
             -- if there's no backup of the file yet, create one
-            if not utils.file_exists(stored_path) then
+            if (not utils.file_exists(stored_path)) then
                 os.execute('cp '..standard_path..' '..stored_path)
+                if (utils.file_exists(remote_path)) then
+                    for key, section in pairs(stored:get_all(file)) do
+                        -- check if section is in remote configuration
+                        local section_check = check:get(file, section['.name'])
+                        if section_check ~= nil then
+                            -- check if options is in remote configuration
+                            for option, value in pairs(section) do
+                                if not utils.starts_with_dot(option) then
+                                    local option_check = check:get(file, section['.name'], option)
+                                    if option_check ~= nil then
+                                        -- if option is in remote configuration, remove it
+                                        stored:delete(file, section['.name'], option)
+                                    end
+                                end
+                            end
+                        end
+                    end
+                end
```

Comment:
I added this section to remove options that are coming from or modified by remote configuration

Reply:
Fixed :sweat_smile: